central/imageintegration/service/service_impl.go

```diff
@@ -332,6 +343,27 @@ func (s *serviceImpl) validateIntegration(ctx context.Context, request *storage.
 }
 
 func (s *serviceImpl) reconcileUpdateImageIntegrationRequest(ctx context.Context, updateRequest *v1.UpdateImageIntegrationRequest) error {
+	var integration *storage.ImageIntegration
+	var exists bool
+	var err error
+
+	// Use the presence of ID to determine if this request is for an updated integration.
+	if updateRequest.GetConfig().GetId() != "" {
+		integration, exists, err = s.datastore.GetImageIntegration(ctx, updateRequest.GetConfig().GetId())
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return errors.Wrapf(errox.NotFound, "image integration %s not found", updateRequest.GetConfig().GetId())
```

Comment:
nit (typically when printing user-provided input we use `%q` as formatter):
```suggestion
			return errors.Wrapf(errox.NotFound, "image integration %q not found", updateRequest.GetConfig().GetId())
```